src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -282,14 +288,23 @@ public JobModel mapFlightStateToJobModel(FlightState flightState) {
       completedDate = flightState.getCompleted().get().toString();
     }
 
-    return new JobModel()
-        .id(flightState.getFlightId())
-        .className(flightState.getClassName())
-        .description(description)
-        .jobStatus(jobStatus)
-        .statusCode(statusCode.value())
-        .submitted(submittedDate)
-        .completed(completedDate);
+    var jobModel =
+        new JobModel()
+            .id(flightState.getFlightId())
+            .className(flightState.getClassName())
+            .description(description)
+            .jobStatus(jobStatus)
+            .statusCode(statusCode.value())
+            .submitted(submittedDate)
+            .completed(completedDate)
+            .iamResourceId(iamResourceId);
+    if (iamResourceType != null) {
+      jobModel.iamResourceType(iamResourceType.getSamResourceName());
+    }
```

Comment:
This is a case where I think `Optional.ofNullable()` can help. Using it avoids duplicating the expression you're checking for `!= null`. Also, I think it's clearer to use the non-fluent version of the setter for a non-fluent usage like this.
```suggestion
    Optional.ofNullable(iamResourceType)
      .map(IamResourceType::getSamResourceName)
      .forEach(jobModel::setIamResourceType);
```

Reply:
Regarding the Optional, this is a style that hasn't caught on consistently through the repo, and in this service the existing convention is to perform null checks.  Optional usage like this can at times make code more difficult for a wider audience to understand, in this case the duplication is worth it to me.

> Also, I think it's clearer to use the non-fluent version of the setter for a non-fluent usage like this.
I agree with you, a later commit made that change.